articles/tokens/refresh-token.md

```diff
@@ -1,152 +1,173 @@
+Refresh tokens can be [obtained](#obtain-a-refresh-token) or [revoked](#revoke-a-refresh-token-with-a-request) programmatically through the Auth0 API.
 
-They can also be viewed and revoked [from the dashboard](#revoke-a-refresh-token-in-the-dashboard).
+They can also be viewed and revoked [from the dashboard](#revoke-a-refresh-token-using-the-dashboard).
 
-## Obtain a Refresh Token
+## Get a Refresh Token
 
-To obtain a refresh token, the `offline_access` scope (see: [Scopes](/scopes)) and an arbitrary `device` name must be included when initiating an authentication request through the [authorize](/api/authentication/reference#authorize-client) endpoint.
+To get a refresh token, you must include the `offline_access` [scope](/scopes) when you initiate an authentication request through the [authorize](/api/authentication/reference#authorize-client) endpoint.
 
 For example:
 
-```text
+```
 GET https://${account.namespace}/authorize/?
     response_type=token
+    &audience=YOUR_API_IDENTIFIER
     &client_id=${account.clientId}
-    &redirect_uri=YOUR_CALLBACK_URL
+    &redirect_uri=${account.callback}
     &state=VALUE_THAT_SURVIVES_REDIRECTS
     &scope=openid%20offline_access
-    &device=my-device
 ```
 
-**NOTE**: The `device` parameter can be any value, such as a unique mobile device identifier.
-
-When the authentication flow completes, Auth0 will redirect the user to the `callback_URL` as usual.
+Once the user authenticates successfully, the client will be redirected to the `callback_URL`.
 The complete URL will be as follows:
 
-```text
-GET https://YOUR_CALLBACK_URL#
+```
+GET ${account.callback}#
     access_token=2nF...WpA
     &id_token=eyJhb...
     &state=VALUE_THAT_SURVIVES_REDIRECTS
     &refresh_token=Cqp...Mwe
 ```
 
-The refresh token is returned as part of the URL, in the form of an opaque string.
+The refresh token (an opaque string) is part of the URL. You should store it securely and use it only when needed.
+
+**NOTE**: In this example, the token was returned to the client in the URL because the [implicit grant](/api-auth/grant/implicit) (`response_type=token`) was used.
 
-**NOTE**: In this case, the token was returned to the client directly in the URL because the [implicit flow](/protocols#oauth2-implicit-flow) (`response_type=token`) was used.
+::: panel-info Troubleshooting
+If the response did not include a refresh token, check the **Allow Offline Access** switch on your [API Settings](${manage_url}/#/apis). It must be enabled, otherwise Auth0 won't return a refresh token, even if you included the `offline_access` scope.
+:::
 
 ## Use a Refresh Token
 
-To obtain a new `id_token`, call the [delegation](/api/authentication/reference#delegation) endpoint in the Authentication API:
+To refresh your token, using the `refresh_token` you already got during authorization, make a `POST` request to the `/oauth/token` endpoint in the Authentication API, using `grant_type=refresh_token`.
```

Comment:
Let's add a warning in this section saying that refresh tokens won't run rules, that's something we will add on the future.